crates/topos-tce-storage/src/validator/mod.rs

```diff
@@ -356,24 +359,24 @@ impl ValidatorStore {
         // For every local known subnets we want to iterate and check if there
         // is a delta between the from_position and our head position.
         for (subnet, local_position) in subnets {
-            let entry = from_positions.entry(subnet).or_default();
-
-            let certs: Vec<_> = if let Some(position) = entry.pop() {
+            let certs: Vec<_> = if let Some(position) = from_positions.get(&subnet) {
                 if local_position <= position.delivery_position.position {
                     continue;
                 }
+
                 self.fullnode_store
                     .perpetual_tables
                     .streams
-                    .prefix_iter_at(&subnet, &position)?
+                    .prefix_iter(&(&subnet, &position.delivery_position.position))?
+                    .skip(1)
```

Comment:
Is this the "hack" we need to fix the bug? Have you figured out exactly why rocksdb behaves weirdly with `prefix_iter_at`? At Parity many years back we had issues with rocksdb iterators behaving in unexpected ways but I have forgotten the details tbh.

Reply:
It is due to how we insert the prefix (`bincode` encoded fixed size int of subnet + position) and how we seek the key.